docs: add documentation about set up with Patchwork

rsarky opened the pull request.

rsarky committed:
docs: add documentation about set up with Patchwork

rralf commented:
Thanks, will test it on monday. We should take this patch to the wip-patchwork-integration branch.

rsarky commented:
This PR is submitted to that branch itself. Do you want me to send you a patch instead?

rsarky committed:
pypasta: config: add patch groups cache

rsarky committed:
wip script to import patches into patchwork from public inboxes

rralf commented on README.md:
```diff
@@ -248,3 +248,40 @@ To compare all mails on the list against upstream:
+1. Install Patchwork on your system, following the guidelines in Patchwork's [documentation](https://patchwork.readthedocs.io/en/latest/development/installation/)
+
+2. Start a shell inside Patchwork's docker container with `docker-compose run --rm web --shell`
+
```
Could you please add two more words how to:
  - create a test project in patchwork
  - import parts of initial archive
  - import the rest of the archive

rsarky replied:
Sure

rralf commented on parse-pubin.py:
```diff
@@ -0,0 +1,38 @@
+        # subprocess.call(['python3', 'manage.py', 'parsearchive'] + parsemail_args + ['publicinboxtmp'])
+with open('publicinboxtmp', 'wb') as f:
+    f.write('\n'.join(messages).encode('utf-8'))
+subprocess.call(['python3', 'manage.py', 'parsearchive'] + parsemail_args + ['publicinboxtmp'])
```
I bet this scripts works, but please, no, don't do it in this way. It's better to have a simple shell script. Just sent you an example per mail. :-)

rsarky replied:
Got it. This was just a hacky thing I set up using a script that Daniel wrote reference.
Will eventually clean this up and make it into a shells script

rsarky committed:
pypasta: patchwork: download patches from Patchwork based on patch id

Signed-off-by: Rohit Sarkar <rohitsarkar5398@gmail.com>

rsarky committed:
wip: sync: change logic to pull patches from patchwork

Currently works with only a single patchwork project.
An initial mbox of the patchwork project needs to be present in the
mbox/raw directory. The name of the mbox should be the same as the
filename defined in the config.

Signed-off-by: Rohit Sarkar <rohitsarkar5398@gmail.com>
